### ISSUE TITLE
Setting UMAP random seed seems to break the model results

### ISSUE THREAD

**Issue by serenalotreck** [MaartenGr/BERTopic#1737]
I have a dataset of scientific abstracts. When I run the following code, I get ~65 topics:

```
sentence_model = SentenceTransformer('allenai/scibert_scivocab_cased')
topic_model = BERTopic(embedding_model=sentence_model)
topics, probs = topic_model.fit_transform(docs)
```

However, if I try defining a random seed in the UMAP model, I get only 2 topics and the outliers, no matter how I set the random seed:

```
umap_model = UMAP(random_state=1234)
sentence_model = SentenceTransformer('allenai/scibert_scivocab_cased')
topic_model = BERTopic(embedding_model=sentence_model, umap_model=umap_model)
topics, probs = topic_model.fit_transform(docs)
``` 

This seems very weird to me; I wouldn't expect the random seed to have such a large effect on the model; but even if it does, I would expect different results when I change the value of the random seed, rather than the difference being between no random seed, and any random seed.

Is this expected or is something weird going on?

Thanks!

**Comment by MaartenGr** [comment]
You did not use the same UMAP parameters as BERTopic is doing by default. Check the init code of BERTopic and you will find what the default parameters are.

**Comment by serenalotreck** [comment]
That fixed it, thank you! Didn't realize those weren't the UMAP defaults.
